src/cli.rs

```diff
@@ -331,11 +331,38 @@ impl Opt {
                 process::exit(1);
             }
         };
-        let root = root.unwrap_or_else(|| home.join(".sheldon"));
-        let config_file = config_file.unwrap_or_else(|| root.join("plugins.toml"));
-        let lock_file = lock_file.unwrap_or_else(|| root.join("plugins.lock"));
-        let clone_dir = clone_dir.unwrap_or_else(|| root.join("repos"));
-        let download_dir = download_dir.unwrap_or_else(|| root.join("downloads"));
+
+        let xdg_config_user = std::env::var_os("XDG_CONFIG_HOME").map(PathBuf::from);
+        let xdg_data_user = std::env::var_os("XDG_DATA_HOME").map(PathBuf::from);
+        let xdg_cache_user = std::env::var_os("XDG_CACHE_HOME");
+
+        let using_xdg =
+            xdg_data_user.is_some() || xdg_config_user.is_some() || xdg_cache_user.is_some();
+
+        let xdg_config = xdg_config_user
+            .or_else(|| home.join(".config").into())
+            .map(|dir| dir.join("sheldon"))
+            .filter(|_| using_xdg);
```

Comment:
I think that doing the check here makes it less clear that there are only really two code paths: XDG directory structure or default structure. It might be clearer to have an if statement and use the `config_dir` and `data_dir` terminology to set the `Settings` values.

```rust
let (config_dir, data_dir) = if using_xdg {
   // ($XDG_CONFIG_HOME/sheldon, $XDG_DATA_HOME/sheldon)
} else {
   // ($HOME/.sheldon, $HOME/.sheldon)
}

// now set directories based on config_dir and data_dir instead of root
let config_file = config_file.unwrap_or_else(|| config_dir.join("plugins.toml"));
// etc ...
let clone_dir = clone_dir.unwrap_or_else(|| data_dir.join("repos"));

let root = data_dir;


```